jscomp/core/js_dump.ml

```diff
@@ -766,6 +766,14 @@ and expression_desc cxt ~(level : int) f x : cxt =
       in
       if p.num_nonconst = 1 && not !Js_config.debug then
         pp_comment_option f (Some p.name);
+      let objs = match p.optional_labels with
+      | [] -> objs
+      | _ ->
+        Ext_list.array_list_filter_map p.fields el (fun f x ->
```

Comment:
This throws away the existing `objs` and rebuilds it from scratch. Not clear it rebuilds it the same way. And if it does, then there are 2 copies of `objs`-building code that need to stay in sync.
Cleanup?

Reply:
Make sense to me. Let me fix it.

Reply:
Fixed https://github.com/rescript-lang/rescript-compiler/pull/5827/commits/a8abd47c8d407821a46d9977e9cd88ee3fedc790
How does it look?